hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/store/wal/WALProcedureStore.java

```diff
@@ -848,14 +850,24 @@ private void syncLoop() throws Throwable {
                       StringUtils.humanSize(syncedPerSec)));
           }
 
-          // update webui circular buffers (TODO: get rid of allocations)
-          final SyncMetrics syncMetrics = new SyncMetrics();
+          // update webui circular buffers
+          SyncMetrics syncMetrics = null;
+          if (syncMetricsQueue.isAtFullCapacity()) {
+            if (syncMetricsQueueIndex == syncMetricsQueueSize) {
+              syncMetricsQueueIndex = 0;
+            }
+            syncMetrics = syncMetricsQueue.get(syncMetricsQueueIndex);
+            syncMetricsQueueIndex ++;
```

Comment:
should be `syncMetricsQueueIndex++; `?

Reply:
Yes, syncMetricsQueueIndex ++ in order reuse next SyncMetrics Object in syncMetricsQueue.